core/domain/story_fetchers.py

```diff
@@ -374,6 +443,58 @@ def get_completed_nodes_in_story(
     return completed_nodes
 
 
+def get_user_progress_in_story_chapters(
+    user_id: str, story_ids: List[str]
+) -> List[story_domain.StoryChapterProgressSummaryDict]:
+    """Returns the progress of multiple users in multiple chapters.
+
+    Args:
+        user_id: str. The user id of the user.
+        story_ids: list(str). The ids of the stories.
+
+    Returns:
+        list(StoryChapterProgressSummaryDict). The list of the progress
+        summaries of the user corresponding to all stories chapters.
+    """
+    all_valid_story_nodes: List[story_domain.StoryNode] = []
+    for story in get_stories_by_ids(story_ids):
+        if story:
```

Comment:
Typically "if story is not None", here and elsewhere.

Reply:
Done!